From 0009853c7dd9a4c57042f647ffc5f5ab734d266f Mon Sep 17 00:00:00 2001
From: YOSHIOKA Takuma <tashioka.256@gmail.com>
Date: Tue, 23 Jan 2018 00:04:09 +0900
Subject: [PATCH 1/2] Inherit typing rule for descendant states (used during
 dict edit)

---
 libskk/context.vala | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libskk/context.vala b/libskk/context.vala
index 3d7e01b..2b02202 100644
--- a/libskk/context.vala
+++ b/libskk/context.vala
@@ -352,6 +352,7 @@ namespace Skk {
 
         void start_dict_edit (string midasi, bool okuri) {
             var state = new State (_dictionaries);
+            state.typing_rule = typing_rule;
             state.midasi = midasi;
             push_state (state);
             update_preedit ();
