diff -Naurp libskk-1.0.2.orig/libskk/key-event.vala libskk-1.0.2/libskk/key-event.vala
--- libskk-1.0.2.orig/libskk/key-event.vala	2018-01-21 00:40:00.541005620 +0900
+++ libskk-1.0.2/libskk/key-event.vala	2018-01-21 17:55:13.931836047 +0900
@@ -237,8 +237,9 @@ namespace Skk {
             { Keysyms.KP_Page_Up, "Page_Up" },
             { Keysyms.Page_Down, "Page_Down" },
             { Keysyms.KP_Page_Down, "Page_Down" },
-            { Keysyms.Muhenkan, "lshift" },
-            { Keysyms.Henkan, "rshift" }
+            { Keysyms.Muhenkan, "Muhenkan" },
+            { Keysyms.Henkan, "Henkan" },
+            { Keysyms.Escape, "Escape" }
         };
 
         /**
